linalg/amgxsolver.hpp

```diff
@@ -313,6 +318,8 @@ class AmgXSolver : public Solver
    int64_t mat_local_rows;
 
    std::string mpi_gpu_mode;
+
+   bool zero_guess = true;
```

Comment:
Following @tzanio 's comment, can you set the `zero_guess` value based on `amgxMode` in `DefaultParameters()` method so it's `false` for a solver and `true` for a preconditioner?